File: Perl_Python/functional/func.py
import re

arr = range(10)

odd = filter(lambda x: not x%2, arr)
even = filter(lambda x: re.search('[13579]$', str(x)), arr)
print odd
print even

squarr = map(lambda x: x**2, arr)
print squarr

print reduce(lambda x,y: x+y, squarr) # Should use sum here

File: Perl_Python/functional/func.pl
use strict;

$\ = "\n";

my @arr = (0..9);

my @even = grep { not $_ % 2 } @arr;
my @odd = grep /[13579]$/, @arr;

print join ', ', @even;
print join ', ', @odd;

my @squarr = map {$_**2} @arr;
print join ', ', @squarr;

use List::Util qw(reduce sum);
print reduce {$a + $b} @squarr; # should use sum here too

